docs/proposals/corss-cluster-statefulset/README.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Your short, descriptive title
+authors:
+- "@robot" # Authors' github accounts here.
+reviewers:
+- "@robot"
+- TBD
+approvers:
+- "@robot"
+- TBD
+
+creation-date: yyyy-mm-dd
+
+---
+
+# Your short, descriptive title
+
+<!--
+This is the title of your KEP. Keep it short, simple, and descriptive. A good
+title can help communicate what the KEP is and should be considered as part of
+any review.
+-->
+
+## Summary
+
+With the development of multi-cluster technology, some users begin to want to migrate stateful applications to multi-cluster scenarios, which cannot be covered well by the current version of karmada. Each user needs to implement a `karmada-operator` that does something similar, so we can propose an API to handle the common coordination related work logic, while setting aside a mechanism for the user's `karmada-operator` to complete the specific logic, such as rolling upgrades, scaling, etc.
+
+## Motivation
+
+Provides a new API for stateful applications across clusters, enabling karmada to adapt to stateful applications across clusters.
+
+An obvious advantage is that karmada natively supports cross-cluster Statefulset, preventing each user from repeatedly implementing his own cross-cluster Statefulset.
+
+### Goals
+
+- Defining an API(crossClusterStatefulset) enables users to implement specific logic for stateful operator applications across clusters.
+-  Propose the implementation ideas for involved components, including the new controller of `cross_cluster_statefulset_controller` in karmada-controller-manager.
+
+
+
+### Non-Goals
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+#### As a user, I want to deploy a cross cluster stateful app.
+
+In order to enable my stateful application to achieve disaster recovery in multiple clusters and obtain better regional services,I want to let my kubernetes operator from member cluster can working for cross cluster stateful with karmada and not need to implement other operator for karmada.
```

Comment:
I think we should give some specific scenarios, then we can see what we expect.

Like MySQL statefulset in multiple clusters, the start index should be different(mysql-0, mysql-1), then we expect only one master MySQL instance. And these instances will detect others' through mysql-{x}.default.xxx.